packages/x-charts-vendor/.babelrc.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Transform d3 ESM libraries to vendored CommonJS libraries
+ *
+ * This produces `lib-vendor/d3-<package name>/src` files that have
+ * internally consistent references to other d3 packages. It is only meant
+ * to be used for the CommonJS import path.
+ */
+const path = require('path');
+
+module.exports = {
+  only: ['node_modules/*/src/**/*.js', 'node_modules/*/**/*.js'],
+  plugins: [
+    [
+      '@babel/transform-modules-commonjs',
+      {
+        strict: false,
+        allowTopLevelThis: true,
+      },
+    ],
+    [
+      'module-resolver',
+      {
+        // Convert all imports for _other_ d3 dependencies to the relative
+        // path in our vendor package.
+        resolvePath(sourcePath, currentFile) {
+          const d3pattern =
+            /^(?<pkg>(d3-[^\/]+|internmap|delaunator|robust-predicates))(?<path>.*)/;
+          const match = d3pattern.exec(sourcePath);
+          if (match) {
+            // We're assuming a common shape of d3 packages:
+            // - Only top level imports "d3-<whatever>"
+            // - With no path components (like "d3-<whatever>/path/to.js")
+            if (match.groups.path) {
+              throw new Error(`Unable to process ${sourcePath} import in ${currentFile}`);
+            }
+
+            // Get Vendor package path.
+            const vendorPkg = ['delaunator', 'robust-predicates'].includes(match.groups.pkg)
+              ? path.resolve(__dirname, `./lib-vendor/${match.groups.pkg}/index.js`)
+              : path.resolve(__dirname, `./lib-vendor/${match.groups.pkg}/src/index.js`);
```

Comment:
Here is the main difference with the original script.

`d3-delaunay` requires `delaunator` which requires `robust-predicates` which seems to be ESM only too.

So I added the to the vendor too.

Those lines are here because they do not have the exact same structure as the d3 packages